Update bitstream format registry

alanorth opened the pull request:
## Description
The DSpace bitstream format registry has not been updated with new image formats in several years, despite new image formats like WebP and AVIF gaining traction on the web. Also, the short description for the PNG format is erroneously `image/png`, which is its mime type, not its human-readable description.

This opens up the door to supporting this image formats more widely in DSpace.

List of changes in this PR:
* Change short description for PNG from "image/png" to "PNG" and update `dspace.cfg` to match
* Add WebP and AVIF image formats

## References:

- [Comparing image formats. JPEG vs WebP vs AVIF (2020)](https://fronius.me/articles/2020-10-14-comparing-image-formats-jpg-webp-avif.html)
- [Comparing AVIF vs WebP file sizes at the same DSSIM (2020)](https://www.ctrl.blog/entry/webp-avif-comparison.html)
- [Using Modern Image Formats: AVIF And WebP (2021)](https://www.smashingmagazine.com/2021/09/modern-image-formats-avif-webp/)

alanorth committed:
dspace: Change description for image/png to "PNG"

"image/png" is the mime type, not the short description.

alanorth committed:
dspace.cfg: update name of PNG format

We should use the short description, not the mime type, as with all
other image types.

alanorth committed:
dspace: add WebP and AVIF to bitstream format registry

WebP and AVIF are modern image formats that have been around for a
few years and have become increasingly widely supported.

tdonohue commented on dspace/config/registries/bitstream-formats.xml:
```diff
@@ -201,7 +201,7 @@
   <bitstream-type>
     <mimetype>image/png</mimetype>
-    <short_description>image/png</short_description>
+    <short_description>PNG</short_description>
```
@alanorth : While I agree this short_description has been wrong for a long time... I'm a little worried about making this change _without_ some sort of migration to automatically update everyone's database entries to use "PNG" instead of "image/png".

Without the migration in place, I'm worried that the Media Filter changes you've made in `dspace.cfg` will no longer work for sites which still have "image/png" instead of "PNG".

So, I think we should either..
1. Create a database migration to migrate all sites from "image/png" to "PNG" in the bitstream format table (where it currently exists)
2. OR, we should undo this change and simply leave it as "image/png" for now....even though it's slightly wrong.

alanorth replied:
Good point, Tim! I think it's worth fixing the have this be correct instead of carrying around legacy typos forever. I will look into creating a database migration to update `image/png` for existing users since the bitstream format registry is only run on new installs.

alanorth replied:
I've added a SQL migration for PostgreSQL, Oracle, and H2, though I have only tested it on PostgreSQL.

alanorth committed:
dspace-api: Add SQL migration for PNG

Add a Flyway SQL migration to update the description of PNG in the
bitstream format registry.

alanorth commented:
After discussion with @tdonohue I've added a Flyway SQL migration to update the bitstream format registry for existing installations, since DSpace only imports the registry from XML on initial install.

alanorth committed:
dspace-server-webapp: Update IT for bitstream format registry

Since we added WebP and AVIF we need to increase the expected numb-
er of formats by two.

alanorth commented:
I had fixed the integration tests for this before #8705 was merged but now I am not sure why they don't pass. Haven't had time to dig into it and the tests take forever to run!

tdonohue commented:
@alanorth : The errors are these type...it's a small issue in counting the number of formats:
```
Error:  Failures: 
Error:  org.dspace.app.rest.BitstreamFormatRestRepositoryIT.createAlreadyExisting(org.dspace.app.rest.BitstreamFormatRestRepositoryIT)
Error:    Run 1: BitstreamFormatRestRepositoryIT.createAlreadyExisting:267 JSON path "$.page.totalElements"
Expected: is <87>
     but: was <85>
```

The easiest way to run individual failing tests locally is to run a *single* test class.  You are correct that running all takes a while...but it's relatively quick to run only one test class (<5mins usually).  The docs on how to do so are in the README: https://github.com/DSpace/DSpace/blob/main/README.md#running-tests

For this situation, if you wanted to debug this individual `BitstreamFormatRestRepositoryIT` class, you should be able to run:
```
mvn install -DskipIntegrationTests=false -Dit.test=BitstreamFormatRestRepositoryIT -DfailIfNoTests=false
```

That will run the full DSpace build process, but only run that *single* test class.

alanorth commented:
@tdonohue perfect, thank you! Running that single test locally is much faster. I fixed the test now after realizing that it is only testing public bitstream formats, not internal ones:

```console
$ grep -c '<internal>false</internal>' dspace/config/registries/bitstream-formats.xml
84
$ grep -c '<internal>true</internal>' dspace/config/registries/bitstream-formats.xml
2
```

tdonohue reviewed (approved):
👍 Thanks @alanorth !  Tested this, including the migration script and it's all working well.
